### ISSUE TITLE
Build fails on macOS(arm64)

### ISSUE THREAD

**Issue by kandizzy** [risc0/risc0#82]
I am experiencing an issue with building on a Mac Mini M1. `bazelisk test //...` results in the error detailed below.

## Steps to Reproduce

I'm not sure how to reproduce this error. The build was successful on a Macbook Air with similar specs listed below.

## Environment 

* macOS Monterey 12.3.1 (21E258)
* Xcode Version 13.3.1 (13E500a)
* Apple clang version 13.1.6 (clang-1316.0.21.2.3)
* Target: arm64-apple-darwin21.4.0

## Logs

```INFO: Reading 'startup' options from /path/to/risc0/.bazelrc: --windows_enable_symlinks
INFO: Options provided by the client:
  Inherited 'common' options: --isatty=1 --terminal_columns=175
INFO: Reading rc options for 'test' from /path/to/risc0/.bazelrc:
  Inherited 'build' options: -c opt --announce_rc --enable_runfiles --nobuild_python_zip --incompatible_enable_cc_toolchain_resolution --flag_alias=accel=//risc0/zkp/accel:flag --//bazel/rules/clang_format:config=//:.clang-format --@rules_rust//:rustfmt.toml=//bazel/rules/rust:rustfmt.toml
INFO: Repository crates_host instantiated at:
  /path/to/risc0/WORKSPACE:109:18: in <toplevel>
Repository rule crates_repository defined at:
  /private/var/tmp/_bazel_ck/23e37feca6eb160088e3173bb952b18e/external/rules_rust/crate_universe/private/crates_repository.bzl:102:36: in <toplevel>
ERROR: An error occurred during the fetch of repository 'crates_host':
   Traceback (most recent call last):
	File "/private/var/tmp/_bazel_ck/23e37feca6eb160088e3173bb952b18e/external/rules_rust/crate_universe/private/crates_repository.bzl", line 24, column 34, in _crates_repository_impl
		host_triple = get_host_triple(repository_ctx)
	File "/private/var/tmp/_bazel_ck/23e37feca6eb160088e3173bb952b18e/external/rules_rust/cargo/private/cargo_utils.bzl", line 95, column 38, in get_host_triple
		cpu = _query_cpu_architecture(repository_ctx, supported_architectures["macos"])
	File "/private/var/tmp/_bazel_ck/23e37feca6eb160088e3173bb952b18e/external/rules_rust/cargo/private/cargo_utils.bzl", line 33, column 13, in _query_cpu_architecture
		fail(_CPU_ARCH_ERROR_MSG.format(
Error in fail: Command failed with exit code '1': ["uname", "-m"]
----------stdout:

----------stderr:
src/main/tools/process-wrapper-legacy.cc:80: "execvp(uname, ...)": No such file or directory

ERROR: /path/to/risc0/WORKSPACE:109:18: fetching crates_repository rule //external:crates_host: Traceback (most recent call last):
	File "/private/var/tmp/_bazel_ck/23e37feca6eb160088e3173bb952b18e/external/rules_rust/crate_universe/private/crates_repository.bzl", line 24, column 34, in _crates_repository_impl
		host_triple = get_host_triple(repository_ctx)
	File "/private/var/tmp/_bazel_ck/23e37feca6eb160088e3173bb952b18e/external/rules_rust/cargo/private/cargo_utils.bzl", line 95, column 38, in get_host_triple
		cpu = _query_cpu_architecture(repository_ctx, supported_architectures["macos"])
	File "/private/var/tmp/_bazel_ck/23e37feca6eb160088e3173bb952b18e/external/rules_rust/cargo/private/cargo_utils.bzl", line 33, column 13, in _query_cpu_architecture
		fail(_CPU_ARCH_ERROR_MSG.format(
Error in fail: Command failed with exit code '1': ["uname", "-m"]
----------stdout:

----------stderr:
src/main/tools/process-wrapper-legacy.cc:80: "execvp(uname, ...)": No such file or directory

INFO: Reading rc options for 'test' from /path/to/risc0/.bazelrc:
  Inherited 'build' options: -c opt --announce_rc --enable_runfiles --nobuild_python_zip --incompatible_enable_cc_toolchain_resolution --flag_alias=accel=//risc0/zkp/accel:flag --//bazel/rules/clang_format:config=//:.clang-format --@rules_rust//:rustfmt.toml=//bazel/rules/rust:rustfmt.toml
ERROR: //bazel/rules/clang_format:config :: Error loading option //bazel/rules/clang_format:config: no such package '@crates_host//': Command failed with exit code '1': ["uname", "-m"]
----------stdout:

----------stderr:
src/main/tools/process-wrapper-legacy.cc:80: "execvp(uname, ...)": No such file or directory

```

**Comment by flaub** [comment]
What happens if you run this from a terminal?

```
uname -m
```

I'm on a Mac M1 and I get:

```
$ uname -m
arm64
```

**Comment by kandizzy** [comment]
@flaub confirming I get the same output `arm64`

**Comment by flaub** [comment]
Only other curious thing I see is `/path/to/risc0`. Is your checkout in `/path/to`?

**Comment by kandizzy** [comment]
Yes, `path/to` is something like `/Users/user/Documents`. I tried various places to see if something odd was happening there. The only thing I've found in Bazel documentation that seems similar is [Rule execution failures](https://bazel.build/docs/sandboxing#rule-failures). I tried adding those flags but so far no luck. I'm not sure if I am adding them in the correct place.

I tried removing most things from `.zshrc`. I have more things installed on this machine than the other, conda, node, etc. Same results. A very mysterious problem.
